File: Medium/003-longest-substring-without-repeating-characters/python/solution.test.py
import unittest
from solution import Solution  # Assumes your code is in solution.py

class TestLongestSubstringWithoutRepeatingCharacters(unittest.TestCase):
    def setUp(self):
        self.sol = Solution()

    def test_standard(self):
        """Standard case: abcabcbb Expected output: 3"""
        string = "abcabcbb"
        result = self.sol.lengthOfLongestSubstring(string)
        self.assertEqual(result, 3)

    def test_all_same_characters(self):
        """All Same Characters case: bbbbb Expected output: 1"""
        string = "bbbbb"
        result = self.sol.lengthOfLongestSubstring(string)
        self.assertEqual(result, 1)

    def test_another_standard(self):
        """Another standard case: pwwkew Expected output: 3"""
        string = "pwwkew"
        result = self.sol.lengthOfLongestSubstring(string)
        self.assertEqual(result, 3)

    def test_gold_standard(self):
        """Gold Standard case: abba Expected output: 2"""
        string = "abba"
        result = self.sol.lengthOfLongestSubstring(string)
        self.assertEqual(result, 2)
    
    def test_no_repeating_chars(self):
        """No repeating characters case: abcdefg Expected output: 7"""
        string = "abcdefg"
        result = self.sol.lengthOfLongestSubstring(string)
        self.assertEqual(result, 7)

    def test_space_and_symbol_diversity(self):
        """Space and symbol diversity case: a b c a! Expected output: 5"""
        string = "a b c a!"
        result = self.sol.lengthOfLongestSubstring(string)
        self.assertEqual(result, 4)

    def test_numerical_mixed_case(self):
        """Numerical mixed case: a b c a! Expected output: 6"""
        string = "12312345!"
        result = self.sol.lengthOfLongestSubstring(string)
        self.assertEqual(result, 6)

if __name__ == "__main__":
    unittest.main()

File: Medium/003-longest-substring-without-repeating-characters/javascript/solution.test.js
const { lengthOfLongestSubstring } = require('./solution')

describe('Longest Substring without repeating characters - Javascript', () => {
    test("Standard case: abcabcbb Expected output: 3", () => {
        const string = "abcabcbb"
        const result = lengthOfLongestSubstring(string)
        expect(result)
        self.assertEqual(result).toEqual(3)
    })
    test("All Same Characters case: bbbbb Expected output: 1", () => {
        const string = "bbbbb"
        const result = lengthOfLongestSubstring(string)
        expect(result)
        self.assertEqual(result).toEqual(1)
    })
    test("Another standard case: pwwkew Expected output: 3", () => {
        const string = "pwwkew"
        const result = lengthOfLongestSubstring(string)
        expect(result)
        self.assertEqual(result).toEqual(3)
    })
    test("Gold Standard case: abba Expected output: 2", () => {
        const string = "abba"
        const result = lengthOfLongestSubstring(string)
        expect(result)
        self.assertEqual(result).toEqual(2)
    })
    test("No repeating characters case: abcdefg Expected output: 7", () => {
        const string = "abcdefg"
        const result = lengthOfLongestSubstring(string)
        expect(result)
        self.assertEqual(result).toEqual(7)
    })
    test("Space and symbol diversity case: a b c a! Expected output: 4", () => {
        const string = "a b c a!"
        const result = lengthOfLongestSubstring(string)
        expect(result)
        self.assertEqual(result).toEqual(4)
    })
    test("Numerical mixed case: a b c a! Expected output: 6", () => {
        const string = "12312345!"
        const result = lengthOfLongestSubstring(string)
        expect(result)
        self.assertEqual(result).toEqual(6)
    })
    
})
